### PR TITLE
Fix Console API getCallerFuncLocation

### LINKED ISSUE #497: Node-inspector hang for 90+ seconds and could not step through code
When set up break point on line 11(line after request.on('data'..)  and line 20 of the following code, it hit line 11, watch and step not working. when "continue", it hangs about 90+ seconds and pause on line 11 on with some data.

Node inspector version : 0.8.1
Node version: 0.10.32
Tried on Mac and Ubuntu 14.04

Sample application that can be tested with any http post.

```js
var http = require('http');
var querystring = require('querystring');

function processPost(request, response, callback) {
    var queryData = "";
    if(typeof callback !== 'function') return null;

    if(request.method == 'POST') {
        request.on('data', function(data) {
        // Case 1:
        // Expected behavior to have problems with debugging here
        // `data` - is big buffer (read it as big array)
        // From documentation:
        // "Be careful about viewing the contents of Buffer objects, each byte is displayed as
        // an individual array element; for most Buffers this will take too long to render."
            queryData += data;
            if(queryData.length > 1e6) {
                queryData = "";
                response.writeHead(413, {'Content-Type': 'text/plain'}).end();
                request.connection.destroy();
            }
        });

        request.on('end', function() {
        // Case 2:
        // If you have problems with debugging here - this is unexpected behavior
            request.post = querystring.parse(queryData);
            callback();
        });

    } else {
        response.writeHead(405, {'Content-Type': 'text/plain'});
        response.end();
    }
}

http.createServer(function(request, response) {
    if(request.method == 'POST') {
        processPost(request, response, function() {
            //Case 3:
            console.log(request.post);
            // Use request.post here
 
            response.writeHead(200, "OK", {'Content-Type': 'text/plain'});
           response.write("posted data is : " + JSON.stringify(request.post));
           response.end();
        });
    } else {
        response.writeHead(200, "OK", {'Content-Type': 'text/plain'});
        response.end();
    }

}).listen(80);
```

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/7086185-node-inspector-hang-for-90-seconds-and-could-not-step-through-code?utm_campaign=plugin&utm_content=tracker%2F195817&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F195817&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #500: Console message location throws error
Now, to receive message location, we throw error to receive call stack.

This is a bad strategy, because debugger stops on each console.log, if `pause on uncaught error` is enabled.
Also here is an error #497

So we can silently pause execution on console.log receive call stack from `exec_state` and resume execution.

### One problem, I can't imagine how to resume execution =(

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/7113596-console-message-location-throws-error?utm_campaign=plugin&utm_content=tracker%2F195817&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F195817&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #509: console.timeEnd crashes node-inspector
I've recently upgraded my node-inspector version, and it appears that console output is now logged in the inspector. Nice!

However, this particular use-case — involving `console.time` and `console.timeEnd` — crashes node-inspector:

```javascript

var provider = "foo";
var feedid = 1;

console.time('fetching ', provider ,' posts for feedid ' + feedid);

(function someAsyncFunction() {
  setTimeout(function() {
    console.timeEnd('fetching ', provider ,' posts for feedid ' + feedid);
  }, 2000);
})();
```

Complete stack trace:

```

events.js:72
        throw er; // Unhandled 'error' event
              ^
TypeError: Cannot read property '1' of null
    at getCallerFuncLocation (eval at WRAPPED_BY_NODE_INSPECTOR (/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/v8-debug.js:86:15), <anonymous>:63:31)
    at new ConsoleMessage (eval at WRAPPED_BY_NODE_INSPECTOR (/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/v8-debug.js:86:15), <anonymous>:10:20)
    at Console.debug.register.messageId [as timeEnd] (eval at WRAPPED_BY_NODE_INSPECTOR (/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/v8-debug.js:86:15), <anonymous>:103:21)
    at /Users/gillesdemey/Documents/Subatomic/n2-aggregator/routes/feeds.js:70:15
    at /Users/gillesdemey/Documents/Subatomic/n2-aggregator/db/cache.js:177:11
    at /Users/gillesdemey/Documents/Subatomic/n2-aggregator/db/cache.js:144:7
    at try_callback (/Users/gillesdemey/Documents/Subatomic/n2-aggregator/node_modules/redis/index.js:573:9)
    at RedisClient.return_reply (/Users/gillesdemey/Documents/Subatomic/n2-aggregator/node_modules/redis/index.js:661:13)
    at HiredisReplyParser.<anonymous> (/Users/gillesdemey/Documents/Subatomic/n2-aggregator/node_modules/redis/index.js:309:14)
    at HiredisReplyParser.emit (events.js:95:17)
```

I'm using node `0.10.33` and node-inspector `0.8.1`.
<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/7391657-console-timeend-crashes-node-inspector?utm_campaign=plugin&utm_content=tracker%2F195817&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F195817&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by 3y3**
Closes #509
Closes #500
Closes #497

### CODE REVIEW

**Comment by 3y3** [comment]
Review blocked by #511

File: lib/ConsoleAgent.js [hunk 1]
```diff
@@ -135,18 +135,22 @@ ConsoleAgent.prototype.injection = function(require, debug, options) {
...
-      var _location = /\((.*):(\d+):(\d+)\)$/.exec(callerStackLine);
-      location.url = _location[1];
-      location.line = _location[2];
-      location.column = _location[3];
-    }
-    return location;
+    var oldPrepareStackTrace = Error.prepareStackTrace,
+        callerFrame;
+
+    Error.prepareStackTrace = function(error, stack) { return stack; }
```
Comment (bajtos) [inline lib/ConsoleAgent.js]: I'd prefer to not override builtin methods, but as long as this works, I can live with it.

**Comment by bajtos** [comment]
Is it worth adding a unit test?

**Comment by 3y3** [comment]
Test added. Hard Coded handle indexes look badly, but I prefer to not overcomplicate test logic now.

**Comment by bajtos** [comment]
LGTM. :shipit:

**Comment by bajtos** [comment]
I'd like to release this as 0.8.2, any objections?

**Comment by bajtos** [comment]
> I'd like to release this as 0.8.2, any objections?

NVM, I see that you are working on a new release in #510.

**Pull request merged**